halfkp.py

```diff
@@ -27,7 +29,7 @@ def halfkp_idx(is_white_pov: bool, king_sq: int, sq: int, p: chess.Piece):
 def get_halfkp_indices(board: chess.Board):
   # TODO - this doesn't support the factors yet.
   def piece_indices(turn):
-    indices = torch.zeros(INPUTS)
+    indices = torch.zeros(Features.inputs)
     for sq, p in board.piece_map().items():
       if p.piece_type == chess.KING:
         continue
```

Comment:
perhaps these should now be a part of `class Features:`

Reply:
also, the lines https://github.com/glinscott/nnue-pytorch/blob/7793bf6620fcd8003692d755c0e735f8ff5566de/serialize.py#L12-L32 could become a part of the factorizer

Reply:
Good call, done.